src/Code/index.js

```diff
@@ -0,0 +1,229 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core'
+import { createRef, useState } from 'react'
+import nextId from 'react-id-generator'
+import classNames from 'classnames'
+import Highlight, { defaultProps } from 'prism-react-renderer'
+import '@spectrum-css/typography'
+import '@spectrum-css/tooltip'
+import '@adobe/prism-adobe'
+import { ActionButton, defaultTheme, Provider } from '@adobe/react-spectrum'
+import PropTypes from 'prop-types'
+
+const tooltipId = nextId()
+
+const openTooltip = (setIsTooltipOpen) => {
+  setIsTooltipOpen(true)
+  setTimeout(() => {
+    setIsTooltipOpen(false)
+  }, 3000)
+}
+
+const copy = (textarea, document, setIsTooltipOpen) => {
+  textarea.current.select()
+  document.execCommand('copy')
+  openTooltip(setIsTooltipOpen)
+}
+
+const parseMetastring = (metastring) => {
+  const matches = metastring.match(/{.*?(?=})}/g)
+  const options = {}
+  if (matches) {
+    for (let i = 0; i < matches.length; i++) {
+      const option = matches[i].slice(1, -1).split(':')
+      const value = option[1] ? options[1].trim() : ''
+      if (value.toLowerCase() === 'true') {
+        options[option[0]] = true
+      } else if (value.toLowerCase() === 'false') {
+        options[option[0]] = false
+      } else {
+        options[option[0]] = value
+      }
+    }
+  }
+  return options
+}
+
+const RE = /{([\d,-]+)}/
+
+const calculateLinesToHighlight = (meta) => {
+  if (!RE.test(meta)) {
+    return () => false
+  }
+  const lineNumbers = RE.exec(meta)[1]
+    .split(`,`)
+    .map((v) => v.split(`-`).map((x) => parseInt(x, 10)))
+  console.log(lineNumbers)
+
+  return (index) => {
+    const lineNumber = index + 1
+    const inRange = lineNumbers.some(([start, end]) =>
+      end ? lineNumber >= start && lineNumber <= end : lineNumber === start
+    )
+    return inRange
+  }
+}
+
+const Code = ({
+  children,
+  className = '',
+  theme = 'dark',
+  copyButton = true,
+  lineNumbers = true,
+  metastring = '',
+  ...props
+}) => {
+  const [isTooltipOpen, setIsTooltipOpen] = useState(false)
+  const language = className.replace(/language-/, '')
+
+  const shouldHighlightLine = calculateLinesToHighlight(metastring)
+  const options = parseMetastring(metastring)
+  const isCopyButton = options.copy !== undefined ? options.copy : copyButton
+  const isLineNumbers =
+    options.numberLines !== undefined ? options.numberLines : lineNumbers
+
+  return (
+    <Provider
+      theme={defaultTheme}
+      colorScheme={theme}
+      scale='medium'
+      UNSAFE_style={{
+        borderRadius: 'var(--spectrum-global-dimension-size-50)'
```

Comment:
the View component supports setting border-radius so you could remove the UNSAFE_style and wrap the whole thing inside a View

Reply:
If I don't put the border radius on the Provider element then I don't see the rounded corners. If the View wraps the provider or is wrapped by the Provider I see the border radius is set properly on the div it produces but the Provider is still a black rectangle.